awx/main/tests/functional/api/test_job.py

```diff
@@ -210,6 +210,36 @@ def test_disallowed_http_update_methods(put, patch, post, inventory, project, ad
     patch(url=reverse('api:job_detail', kwargs={'pk': job.pk}), data={}, user=admin_user, expect=405)
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "job_type",
+    [
+        'run',
+        'check',
+    ],
+)
+def test_job_relaunch_with_job_type(post, inventory, project, machine_credential, admin_user, job_type):
+    # Create a job template
+    jt = JobTemplate.objects.create(name='testjt', inventory=inventory, project=project)
+
+    # Create a job instance
+    job = jt.create_unified_job()
```

Comment:
a job is created with `job_type` "run" by default, so changing from "run" to "run" doesn't test changing from "run" to "check".
Maybe here we can create a job with the opposite value. That is, if the pytest parameter is "run", create job with job_type "check", and if "check", start with job_type "run".

Reply:
Fixed in new commit - thanks for noticing